cmd/soroban-rpc/lib/preflight/src/fees.rs

```diff
@@ -113,45 +122,54 @@ fn estimate_max_transaction_size_for_operation(
 }
 
 fn calculate_host_function_soroban_resources(
-    snapshot_source: &ledger_storage::LedgerStorage,
-    storage: &Storage,
+    ledger_changes: &Vec<LedgerEntryChange>,
+    pre_storage: &LedgerStorage,
+    post_storage: &Storage,
     budget: &Budget,
     events: &Vec<DiagnosticEvent>,
 ) -> Result<SorobanResources, Box<dyn error::Error>> {
-    let fp = storage_footprint_to_ledger_footprint(&storage.footprint)?;
-    /*
-      readBytes = size(footprint.readOnly) + size(footprint.readWrite)
-      writeBytes = size(storage.map[rw entries])
-      metadataSize = readBytes(footprint.readWrite) + writeBytes + eventsSize
-    */
-    let original_write_ledger_entry_bytes =
-        calculate_unmodified_ledger_entry_bytes(fp.read_write.as_vec(), snapshot_source, false)?;
-    let read_bytes =
-        calculate_unmodified_ledger_entry_bytes(fp.read_only.as_vec(), snapshot_source, false)?
-            + original_write_ledger_entry_bytes;
-    let write_bytes =
-        calculate_modified_read_write_ledger_entry_bytes(&storage.footprint, &storage.map, budget)?;
-    let meta_data_size_bytes =
-        original_write_ledger_entry_bytes + write_bytes + calculate_event_size_bytes(events)?;
+    let ledger_footprint = storage_footprint_to_ledger_footprint(&post_storage.footprint)?;
+    // TODO: check this is correct
+    let mut unmodified_ledger_keys: Vec<LedgerKey> = Vec::new();
+    for k in ledger_footprint.read_only.as_vec() {
+        if !post_storage.map.contains_key::<LedgerKey>(k, budget)? {
+            unmodified_ledger_keys.push(k.clone());
+        }
+    }
+    let unmodified_entries_read_bytes =
+        calculate_unmodified_ledger_entry_bytes(&unmodified_ledger_keys, pre_storage, false)?;
+    let modified_entries_read_bytes: u32 = ledger_changes
+        .iter()
+        .map(|c| c.encoded_key.len() as u32 + c.old_entry_size_bytes)
+        .sum();
```

Comment:
@dmkozh @2opremio thoughts? The `ledger_changes` don't seem to be enough to compute the full `read_bytes` since they don't seem to contain unmodified (but read) ledger entries.

Reply:
That's a bug in recording mode - this hack shouldn't be needed when it's fixed.

Reply:
Good to know, would you please answer here once it's fixed?